float64[] x_ref
---
bool success